code/datums/circuits.dm

```diff
@@ -0,0 +1,117 @@
+#define ALPHA 1
+#define BETA 2
+#define GAMMA 4
+#define DELTA 8
+#define ETA 16
+#define THETA 32
+#define IOTA 64
+
+/datum/circuits
+	var/atom/holder = null //Which specific board are we pointing at?
+	var/total_fuse_points = 7 //You could increase this for more board options. 7 Choose 5 = 21. 9 Choose 5 = 126
+	var/list/fuse_point_names = list("Alpha" = ALPHA, "Beta" = BETA, "Gamma" = GAMMA, "Delta" = DELTA, "Eta" = ETA, "Theta" = THETA, "Iota" = IOTA)
+	var/list/possible_boards = list(/obj/item/weapon/circuitboard/autolathe,/obj/item/weapon/circuitboard/seed_extractor,/obj/item/weapon/circuitboard/conveyor,/obj/item/weapon/circuitboard/air_alarm,/obj/item/weapon/circuitboard/fire_alarm,/obj/item/weapon/circuitboard/airlock,/obj/item/weapon/circuitboard/power_control,/obj/item/weapon/circuitboard/vendomat,/obj/item/weapon/circuitboard/microwave)
+	var/global/list/assigned_boards = list()
+	//Each bitflag points to a board!
+	var/localbit = 0 //What are WE programmed to? Always start as 0
+
+	var/table_options = " align='center'"
+	var/row_options1 = " width='80px'"
+	var/row_options2 = " width='80px'"
+	var/window_x = 240
+	var/window_y = 300
+
+/datum/circuits/New(var/atom/homeboard)
+	..()
+	holder = homeboard
+	if(!(src.assigned_boards.len))
+		generate_schema()
+
+/datum/circuits/proc/generate_schema()
+	for(var/C in possible_boards)
+		var/newbit = newhash()
+		while(!check_config(newbit))
+			newbit = newhash()
+		assigned_boards["[newbit]"] = C
+	return
+
+/datum/circuits/proc/check_config(var/proposed)
+	for(var/bitflag in assigned_boards)
+		if(text2num(proposed) == text2num(bitflag))
+			return 0
+	return 1
+
+/datum/circuits/proc/newhash() //Returns a bitflag
+	var/list/fuse_point_list = list(ALPHA,BETA,GAMMA,DELTA,ETA,THETA,IOTA)

```

Comment:
I suggest you make the list of values itself global at the top of the file, and then copy that value here.
Organisation helps extension.